File: Propeers/Initial-Start-Up/Arrays/Make_Unique_Array.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function minIncrementForUnique(nums) {
    nums.sort(function (a, b) { return a - b; });
    var count = 0;
    for (var i = 1; i < nums.length; i++) {
        if (nums[i] <= nums[i - 1]) {
            count += nums[i - 1] - nums[i] + 1;
            nums[i] = nums[i - 1] + 1;
        }
    }
    return count;
}
;
var nums = [3, 2, 1, 2, 1, 7];
console.log(minIncrementForUnique(nums));

File: Propeers/Initial-Start-Up/Arrays/Make_Unique_Array.java
import java.util.Arrays;
import java.util.Scanner;

public class Make_Unique_Array {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];
        for (int i = 0; i < size; i++) {
            System.out.printf("Enter %d number element of the array: ", i + 1);
            arr[i] = input.nextInt();
        }
        input.close();
        System.out.println(minIncrementForUnique(arr));
    }

    private static int minIncrementForUnique(int[] nums) {
        // Sort the array first...
        Arrays.sort(nums);
        int count = 0;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] <= nums[i - 1]) {
                count += nums[i - 1] - nums[i] + 1;
                nums[i] = nums[i - 1] + 1;
            }
        }
        return count;
    }
}